### DISCUSSION TITLE
Working across organizations with Github apps

### DISCUSSION THREAD

**Discussion by danielrenaud** [community/community#24549] [Apps, API and Webhooks]
<p>In my understanding; a Github app can be installed in multiple orgs, but it is only possible to be authenticated with a single org at a time. With a script or application you can work with multiple orgs be re-authing.</p>
<p>However, given my understanding of authentication is correct, is it possible to perform an action that requires authentication with multiple orgs at the same time. From the situation I’m encountering (described below) it would seem that it is not.</p>
<p>I have two organizations; <code>org1</code> and <code>org2</code>. In each org there is a <strong>private</strong> repo; <code>repo1</code>. <code>org2:repo1</code> is forked from <code>org1:repo1</code>.</p>
<p>I want to create a PR for <code>org2:repo1</code> -&gt; <code>org1:repo1</code>. The app is authenticated with <code>org1</code>.</p>
<p>I always recieve a “422: not all refs are readable” error when trying to create the PR. The app is installed on both orgs and I’ve checked that I can make a PR inside each org.</p>
<p>It seems like the app needs to read from <code>org2</code>, but since the repo is private and the app is not authenticated with that org, it is not able to read the repo and I get this error.</p>
<p>Is this a known limitation or is there some way to work around this?</p>
<p>Does it matter if <code>org2</code> is an actual organization or a personal user account?</p>
<p>Thanks</p>

**Comment by lee-dohm** [comment]
<p>Have you granted “Repository contents” permission (both read and write) to the GitHub app? In order to create a PR, it needs to be able to check that the refs that the PR is created from exist. In order to do that, it needs to be able to inspect the repository contents.</p>
<p>Let us know if that helps!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>We’ve double-checked this with the team working on GitHub Apps and – you’re right. For the case that you described (making a pull request between a private fork and parent) will not work if you’re doing it as a GitHub App. The reason is as you guessed – a single installation has access to repositories only from a specific user or organization, and a pull request between a private fork and parent requires access from two organizations. I’ve opened an internal issue to see if there is a way to support this after all, but even if that’s possible – I’m guessing it won’t happen very quickly since permissions are never a trivial thing to implement.</p>
<p>Other cases (pull requests between public fork and parent, and pull requests within a single private repository) should still work.</p>
<p>Apologies for the spurious question!</p>

**Reply by linker34** [reply]
It seems that this is still an issue, we have the use case where the GitHub Application that we created requires to be installed on multiple organizations to manage the GHA Runners of them and it doesn't work. Hopefully this will be considered soon.

**Comment by PhoenixHe-NV** [comment]
<p>Hi <a class="mention" href="/lee-dohm">@lee-dohm</a> is this issue resolved or in progress? We still get this error while making PR across private fork.</p>

**Comment by ionutzaharia** [comment]
It seems that this is still an issue, we have the use case where the GitHub Application that we created requires to be installed on multiple organizations to manage the GHA Runners of them and it doesn't work. Hopefully this will be considered soon.

**Comment by MikeJerred** [comment]
This is still an issue 4 years later.

**Comment by Jaegin** [comment]
Hi @lee-dohm - is this being actively worked on? And, if so, when can a fix be expected by?

**Comment by roulettedares** [comment]
We need the ability to install a Github App on an entire enterprise instead of at the organization level.

**Comment by mithunkrb** [comment]
> We need the ability to install a Github App on an entire enterprise instead of at the organization level.

Hi Team, we are also requiring this ability to install a Github App on multiple orgs without making it public app. Do we have any PR or issue tracking this? If yes, could you please share?

**Reply by hpsin** [reply]
:wave: This shipped to GA in March, sorry I missed this comment. You can have an `internal` app for your enterprise that is just accessible to your enterprise's orgs. 
https://github.blog/changelog/2025-03-10-enterprise-owned-github-apps-are-now-generally-available/

**Reply by dserodio** [reply]
I read this post and it mentions:
> However, Enterprise Classic organizations and standard user accounts can only transfer private Apps, as internal Apps are not supported in Enterprise Classic.

This is the first time I hear about "Enterprise Classic", and a Google search for `github "enterprise classic"` is not helpful. What is Enterprise Classic and how can I find out if my Enterprise is "Classic" or not?

**Reply by hpsin** [reply]
Ah, sorry! It's a term for non-EMU enterprises, i.e, where you use your personal account to work within the enterprise, as opposed to EMU enterprises. I'll see about making sure that has a hit in the docs.

**Comment by dany74q** [comment]
Same here, are enterprise-scoped apps / tokens on the roadmap ?
It would be extremely useful for integrations in large enterprises.

**Comment by Harry654** [comment]
Hi, please has the issue of creating a PR across forks been resolved?

**Comment by jluque0101** [comment]
Hi friends, is there any update on this?

**Comment by kaovilai** [comment]
I want a GitHub App to "impersonate" and use my identity to comment or review.a PR as me.. as good as PAT.

**Comment by Cavaler** [comment]
Same feature is required if you want to create a repo on one organization from a template in a different one.
Looking forward to see it implemented, though after 6 years not that much hope...

**Comment by ad-zsolt-imre** [comment]
We are facing the same issue. Would need to be able to share GH app across multiple orgs.

**Comment by jasontempleman-eaton** [comment]
Same here.  Any update on this?

**Comment by xmo-odoo** [comment]
FWIW if an application is authorized via a user account, and installed in multiple organizations, the application can create [user access tokens](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-user-access-token-for-a-github-app) (as opposed to installation access tokens) and interact with multiple organizations at the same time.

This does mean the application will be identified as the user (rather than the application), and requires the user account to have access to the repositories with a sufficient level of access, but it does allow using the much finer grained access of github apps (including IP range allow lists etc...)
